File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number}
     */
    goodNodes(root) {
        let count = 0;

        const traverse = (node, max) => {
            if(node === null) return;

            if(node.val >= max) ++count;
            
            const newMax = Math.max(max, node.val)
            traverse(node.left, newMax);
            traverse(node.right, newMax);
        }

        traverse(root, root.val);

        return count;
    }
}

File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        l, r = 0, len(heights) - 1

        max_area = 0
        while(l != r):
            if heights[l] < heights[r]:
                area = heights[l] * (r - l)
                max_area = max(max_area, area)
                l = l + 1
            else:
                area = heights[r] * (r - l)
                max_area = max(max_area, area)
                r = r - 1

        return max_area